src/libraries/System.IO.Hashing/src/System/IO/Hashing/XxHash3.cs

```diff
@@ -0,0 +1,884 @@
+        /// <param name="source">The data to hash.</param>
+        /// <param name="seed">The seed value for this hash computation.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="source"/> is null.</exception>
+        public static byte[] Hash(byte[] source, long seed)
+        {
+#if NET6_0_OR_GREATER
+            ArgumentNullException.ThrowIfNull(source);
+#else
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
+            }
+#endif
+
+            return Hash(new ReadOnlySpan<byte>(source), seed);
+        }
+
+        /// <summary>Computes the XXH3 hash of the provided data using the optionally provided seed.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <param name="seed">The seed value for this hash computation. The default is zero.</param>
+        /// <returns>The XXH3 64-bit hash code of the provided data, written in little-endian order.</returns>
+        public static byte[] Hash(ReadOnlySpan<byte> source, long seed = 0)
+        {
+            byte[] result = new byte[HashLengthInBytes];
+            BinaryPrimitives.WriteInt64LittleEndian(result, HashToInt64(source, seed));
+            return result;
+        }
+
+        /// <summary>Computes the XXH3 hash of the provided data into the provided destination using the optionally provided seed.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <param name="destination">The buffer that receives the computed 64-bit hash code, written in little-endian order.</param>
+        /// <param name="seed">The seed value for this hash computation. The default is zero.</param>
+        /// <returns>The number of bytes written to destination.</returns>
+        public static int Hash(ReadOnlySpan<byte> source, Span<byte> destination, long seed = 0)
+        {
+            if (!BinaryPrimitives.TryWriteInt64LittleEndian(destination, HashToInt64(source, seed)))
+            {
+                ThrowDestinationTooShort();
+            }
+
+            return HashLengthInBytes;
+        }
+
+        /// <summary>Attempts to compute the XXH3 hash of the provided data into the provided destination using the optionally provided seed.</summary>
+        /// <param name="source">The data to hash.</param>
+        /// <param name="destination">The buffer that receives the computed 64-bit hash code, written in little-endian order.</param>
+        /// <param name="bytesWritten">When this method returns, contains the number of bytes written to <paramref name="destination"/>.</param>
+        /// <param name="seed">The seed value for this hash computation. The default is zero.</param>
+        /// <returns>true if destination is long enough to receive the computed hash value (8 bytes); otherwise, false.</returns>
```

Comment:
```suggestion
        /// <returns><see langword="true" /> if <paramref name="destination" /> is long enough to receive the computed hash value (8 bytes); otherwise, <see langword="false" />.</returns>
```